test/util/range.cpp

```diff
@@ -213,6 +213,21 @@ void test_pointers()
     BOOST_CHECK(bgr::at(r2, 5) == 5);
 }
 
+void test_const_pointers()
```

Comment:
To be honest I inspected the tests in [util] only, so there may be other parts of the lib that use more functionality of Boost.Test. Yes, I'm talking about things like test suites/cases with automated registration (e.g BOOST_AUTO_TEST_SUITE( Boost Geometry [util]).

Does the *tests* have to be header only ? That sounds like a weird requirement...

Here is what I have in mind for the Jamefile:

```
project geometry/util/test
   : source-location .
   : requirements
     <source>/boost/test//boost_unit_test_framework
   ;

unit-test range : range.cpp ; 
```

This does not change the header only requirement for Boost.Geometry, but requires to build boost_unit_test_framework for testing (the previous lines are enough to trigger the compilation and link of the framework).